src/app/(BottomTabNavigation)/Profile/EditAddress/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { router } from 'expo-router';
+import React, { useEffect, useState } from 'react';
+import { Text, View, TextInput, TouchableOpacity } from 'react-native';
+
+import styles from './styles';
+import {
+  getCurrentUserInfo,
+  updateCurrUserAddress,
+} from '../../../../supabase/queries/auth';
+
+function EditNameScreen() {
+  const [streetName, setStreetName] = useState<string>();
+  const [usState, setUsState] = useState<string>();
+  const [city, setCity] = useState<string>();
+  const [zip, setZip] = useState<string>();
+  useEffect(() => {
+    getCurrentUserInfo().then(result => {
+      setStreetName(result.streetName);
+      setUsState(result.state);
+      setCity(result.city);
+      setZip(result.zip);
+    });
+  }, []);
+  return (
+    <View>
+      <Text>Edit Address</Text>
+      <TextInput
+        style={styles.input}
+        value={streetName}
+        onChangeText={setStreetName}
+        placeholder="Input Street Name"
+      />
+      <TextInput
+        style={styles.input}
+        value={city}
+        onChangeText={setCity}
+        placeholder="Input City Here"
+      />
+      <TextInput
+        style={styles.input}
+        value={usState}
+        onChangeText={setUsState}
+        placeholder="Input State Here"
+      />
+      <TextInput
+        style={styles.input}
+        value={zip}
+        onChangeText={setZip}
+        placeholder="Input Zip Code Here"
+      />
+      <TouchableOpacity
+        style={styles.button}
+        onPress={() => {
+          if (streetName && city && usState && zip) {
+            updateCurrUserAddress(streetName, city, usState, zip);
```

Comment:
I think this query is ran every time which should not be an issue but let's remove the `if-else` statement since this is the case.

Reply:
right now the if statement is needed in case a user tries to input an empty string as a value. as for checking if the address/ full name field has actually changed, that is something that will be addressed after auth context is created 💃

Reply:
Oh I see!!